### PR TITLE
test-network: check for captive portals received via NDISC 

### PR DESCRIPTION
**Pull request by mrc0mmand**
This requires fairly recent radvd that supports sending RAs with captive
portals [0].

Also, this should hopefully provide coverage for issues like:
  - https://github.com/systemd/systemd/issues/28229
  - https://github.com/systemd/systemd/issues/28231
  - https://github.com/systemd/systemd/issues/28277

[0] https://github.com/radvd-project/radvd/pull/141

### CODE REVIEW

**Comment by mrc0mmand** [comment]
Also a note: there's an accompanying change in the CentOS CI repo - https://github.com/systemd/systemd-centos-ci/commit/ae46e30b2b5d78b4b8dc27862abd63b0c997ea60 - so both Arch Linux jobs run with the latest radvd that supports captive portals.

File: test/test-network/systemd-networkd-tests.py [hunk 1]
```diff
@@ -4553,6 +4579,67 @@ def test_router_preference(self):
...
+        start_radvd(config_file='captive-portal.conf')
+        networkctl_reconfigure('client')
+        self.wait_online(['client:routable'])
+
+        self.wait_address('client', '2002:da8:1:99:1034:56ff:fe78:9a00/64', ipv='-6', timeout_sec=10)
+        output = check_output(*networkctl_cmd, 'status', 'client', env=env)
+        print(output)
+        self.assertIn('Captive Portal: http://systemd.io', output)
+
+        stop_radvd()
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: This is not necessary.

File: test/test-network/systemd-networkd-tests.py [hunk 2]
```diff
@@ -4553,6 +4579,67 @@ def test_router_preference(self):
...
+            start_radvd(config_file='bogus-captive-portal.conf')
+            networkctl_reconfigure('client')
+            self.wait_online(['client:routable'])
+
+            self.wait_address('client', '2002:da8:1:99:1034:56ff:fe78:9a00/64', ipv='-6', timeout_sec=10)
+            output = check_output(*networkctl_cmd, 'status', 'client', env=env)
+            print(output)
+            self.assertNotIn('Captive Portal:', output)
+
+            stop_radvd()
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: Not a matter, but I'd like to move this before `start_radvd()`, as this is for restarting radvd.

File: test/test-network/systemd-networkd-tests.py [hunk 3]
```diff
@@ -695,7 +721,7 @@ def setUpModule():
         [
             '[Service]',
             'ExecStart=',
-            f'ExecStart=!!{udevd_bin}',
+            f'ExecStart=!!@{udevd_bin} systemd-udevd',
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py:724]: Oh, I did not know the prefix `@`. Nice!

**Review by yuwata — APPROVED** [review]

**Pull request merged**